examples/quickstart_etl/README.md

```diff
@@ -0,0 +1,208 @@
+# Dagster starter kit
+
+This example is a starter kit for building a daily ETL pipeline. It showcases Dagster's basic features such as **[Assets](https://docs.dagster.io/concepts/assets/software-defined-assets)** and **[Schedules](https://docs.dagster.io/concepts/partitions-schedules-sensors/schedules)**. It also follows Dagster's recommendations on [using environment variables and secrets](https://docs.dagster.io/guides/dagster/using-environment-variables-and-secrets) to ensure the project is production-ready.
+
+This project is scaffolded with [`dagster project scaffold`](https://docs.dagster.io/getting-started/create-new-project).
```

Comment:
The materializations in this example are pretty magical. On the one hand thats great, users don't have to worry about IO or resources as concepts when viewing this quickstart. On the other hand, it might be a little _too_ magical? 

 Maybe we could leave a breadcrumb in this "Understanding the Code" section that says something along the lines of: 
 
In this example, Dagster defaults to saving each asset to disk. In most projects, your assets will be materialized to a production system, see [some relevant link](...).

Reply:
I intentionally didn't include IO managers or resources in this basic quickstart because I think it may overwhelm beginners. i'll add a section in readme, as well as comments in code to guide people to next steps.

Reply:
i like @slopp's suggestion on 'Understanding the Code'. I'm not sure if we were trying to balance targeting beginners vs people who have dagster knowledge, but if we are, i think we should err heavily on the side of assuming almost no foreknowledge of Dagster. 

1. Is there any helpful reading elsewhere that would be useful to prepare them? for ex. 
_Before you get started, reading 'How Dagster Works [article to be created],'ETL on Dagster [to be created] and [Airflow vs Dagster] to help you understand what we're talking ab_out. 

2. A more comprehensive listing of the features that the user will encounter in the document... for ex. 
_This example is a starter kit for building a daily ETL pipeline. It will show you how to take data from an API and materialize it into a source asset which is transformed over a few steps to materialize an asset output which helps visualize the data.  

In this quick start, you will learn: 
1. That ops (computations) jobs (sequences of computation) and assets (inputs/outputs) work together to transform data.  
3. How to use inputs/outputs on ops to create dependencies that inform Dagster about the relationship between steps 
4. Metadata can be added to your code to improve visualization in the UI (links to asset groups, descriptions, etc) 
5. Transformations can be triggered by a schedule you define separately in python. 
6. How to use Env Vars to store secrets to connect Dagster to other tools
7. How the different files inside a Dagster project are used.
 
To get the most out of this quick start repository, you'll want to look through the files and review the comments we've made for each section to see how things work._ 

I think we'll find that we're generally missing some high level introductions to some concepts that we want to explain more fully, so perhaps the links can be added later.

Reply:
@erinkcochran87 and i are talking about updating the tutorial to use the quickstart script as a nux follow up. 
basically we can make the final outcome of our oss tutorial to be this quick start, and move the heavy product eduction part out of this README to living in tutorial which is a better form for beginners. Concretely, we can add a link to the tutorial at the beginning of this README for users who would like to learn by doing to get started with dagster.

Reply:
Would users need to get a github / hackernews API key to do the tutorial? That might be a source of friction.

Reply:
hn apis don't require any token so a user can run through the tutorial with any extra step outside of dagster, which the main reason why i pivoted from github to hn.

Reply:
oh awesome

Reply:
tracking here: https://github.com/dagster-io/dagster/issues/10604